components/eip681/src/parse.rs

```diff
@@ -0,0 +1,1533 @@
+//! Types and functions used for parsing.
+
+use std::borrow::Cow;
+
+use nom::{AsChar, Parser};
+use snafu::prelude::*;
+
+#[derive(Debug, Snafu)]
+pub enum ParseError<'a> {
```

Comment:
Style: I'd prefer this to go in a separate `src/parse/error.rs` file (like we tend to do in our other crates) so it doesn't get in the way of reading the parser impl.